### ISSUE TITLE
Add a setting to disable kiai note flashing

### ISSUE THREAD

**Issue by mouzedrift** [ppy/osu-stable-issues#390]
I myself find this pretty distracting and it actually hurts my eyes while playing faster maps (especially in taiko). I've also seen lots of people who share the same thoughts and would like to see this as an optional setting.

**Comment by peppy** [comment]
disable the game

**Comment by mouzedrift** [comment]
I'm not trying to be a troll but this seriously hurts my eyes when playing taiko (i don't notice the flashing in other game modes tho). If this is not going to be added to stable could this be added to osu! lazer?

**Comment by peppy** [comment]
Have you tried a different skin? if you reduce the coloured area it should greatly degrease the effect of kiai.

**Comment by mouzedrift** [comment]
Yes, I've also tried to reduce the note transparency, but the note flashing is unfortunately still very instrusive. Therefore, I decided to open this issue to solve the problem (hopefully).

Just to be sure, we're talking about the flashing on the ``taikohitcircle.png`` and not about the colour changes of the ``taiko-bar-right-glow.png`` during kiai times, right?

**Comment by WinterIsPretty** [comment]
https://osu.ppy.sh/help/wiki/Skinning/osu!taiko

My skin has no kiai effect so maybe you can go through the skinning wiki to see what elements cause this

**Comment by jjhhgg100123** [comment]
Curious if this can still be considered, even if it's just included under the existing "Shaders" toggle.

**Comment by mouzedrift** [comment]
additionally this seems to have a pretty annoying side effect on low end GPU's in my case intel hd graphics 4600. normally the game runs at stable 480 fps but when the kiai effect starts my fps become very unstable and start going down 100 fps and then back all the time resulting in a very annoying hitobject speed-up/speed-down in taiko which can be very annoying and hard to hit.

similar result on lazer except my normal fps are around 200 (draw) and they drop around 150 with kiai which seems better overall compared to stable but the base fps are much worse + micro stutters (not related to kiai) resulting in more screen tearing.

hope this can still be considered until lazer can run better than stable on low end GPU's since i wouldn't mind using lazer if it had the same performance as stable or even better.

**Comment by smoogipoo** [comment]
Can you show a video of this happening? I have a hard time believing that kiai can cause such a difference since there's no actual draw overhead afaik, even on stable.

**Comment by mouzedrift** [comment]
https://youtu.be/QoCAKkVMALA here a video on stable i set my obs settings to be pretty minimal so it won't eat many resources while recording since i don't have a working phone atm. first kiai section starts at 0:38 and ends at 1:08. the fps immediately drop in the video every time kiai starts and recover when it ends.

**Comment by smoogipoo** [comment]
Does this happen with the default skin?

**Comment by mouzedrift** [comment]
yep and the fps drop seems to be even worse with the default skin

**Comment by ghost** [comment]
any update on this? please? i seriously sometimes want to quit the game because of kiai flash

**Comment by innynips** [comment]
Any actual reason for this not to be implemented?

**Comment by 3roiler** [comment]
I have the same problem, it's painful for my eyes and I would deeply love an QoL feature to disable the kiai flashing or a option to make it less flashy. It's a visual feedback that shouldn't be fixed/hardcoded, it doesn't change the difficulty of the map, just makes our eyes happier and the osu! community. 
Or it should be implemented in osu!lazer.

There are enough people that would LOVE that feature. There isn't any actual reason to keep it fixed.

BTW the response ..
> disable the game 

.. has a very bad taste of community handling. I know this issue is ~4 years old, but everytime I look in this issue for an update, it's just being ignored and that makes me sad. Really sad.

That's not the way to handle such a NORMAL request. Give us an actual reason to just ignore it, not just bashing at people for a question.

**Comment by peppy** [comment]
I can reword my response if it helps: This is the game's design direction and part of gameplay.

**Comment by 3roiler** [comment]
It does make the game kind of more enjoyable or the gameplay more intense. But my question was to make it customizable. Like a slidebar in the settings to change the intensity of it from a range from 20 - 100% (or disable it, but this is obviously not an option for you).

A slidebar to change like the percentage of the glowing flash would be really, really, really, really, really, really (.. and so on) nice.
Over and out 👋

**Comment by Ashclown** [comment]
I think it would be a really good change if it could be customizable. Please consider this if it's not asking for too much.

**Comment by Ros3sDark** [comment]
I hope the setting will be added soon. its really painful to play like this.

**Comment by OtavioBaltrusch** [comment]
I'll stop chasing this since I noticed there is no amount of reasoning that could possibly convince anyone of this happening, but being a big fan of the development team of this game I'll say I'm fairly disappointed with the way that this accessibility option is being handled.

There doesn't seem to be any reason for this to not be in the game other than "I don't like how it looks without it", while kiai flashing barely adds anything to the game other than making it very uncomfortable to play for people with light sensitivity. Most people who play the game don't even know it's there and probably wouldn't even notice if it was gone, while at the same time it would be a blessing for people who have eye issues.

There are so many accessibility options already in the game but apparently this is where the line is arbitrarily drawn for seemingly no reason. I still have a lot of respect for everyone involved in the game's development as it is a great game with a lot of transparency and players are clearly heard, which makes it all the more saddening to see how unreasonable they can be when dealing with such a minor thing for most people but a huge and unnecessary inconvenience for a few players.

**Comment by peppy** [comment]
The thing is, most every game – and more generally, form of media – (ab)uses light as a mechanic. I would argue that anyone with sensitivity to light can still play the game by addressing it in other ways:

- Make sure you aren't playing against a black background. The human eyes are sensitive to contrast specifically, and you are in a controlled environment here that you can adjust the contrast.
- Play with your monitor settings. It sounds like you may have your gamma, brightness or contrast set in a way that is negatively affecting your ability to enjoy games.


Kiai is not the "drawn line" you may think it is. There are hundreds of other flashes in the game, and I am not interested in removing / adding toggles for each and every one. Adding a toggle for this will only mean users find the next thing to ask for a toggle. We're already down that hole where the game I see people play is nothing like the game I created, nothing like the experience I intended people to play.

And while you may have well a medical condition with light sensitivity, if this option is added, **ONE. HUNDRED. PERCENT.** of "pro" players will turn it off. Arguing that it's not a competitive advantage is, excuse the wording, complete bullshit.

**Comment by OtavioBaltrusch** [comment]
I appreciate the well-versed response, I was hoping to understand the reason why this can't be done and I think I get it, it's probably very sad to have a vision for a game and how you want it to play then have everyone complain about every aspect of it until it's stripped down to the very basic elements. I guess my thought process is that most people don't really notice it as I rarely see anyone talking about this other than people who have an issue with it, but it's quite likely that most "osu! settings guide 2025 (updated)" videos would say to turn this off if it was an option.

Unfortunately I already tried most settings to mitigate this other than the background part, so I'll give that a try and see if it helps, though I might just have to stick to McOsu for playing DT and hope for a day where it doesn't bother me as much to try and play it in Lazer.

I'm still hopeful of revisiting a slider approach to this but I understand if it never happens and that's fine, thank you for the explanation.
